src/library/scala/collection/immutable/Queue.scala

```diff
@@ -148,7 +150,10 @@ sealed class Queue[+A] protected(protected val in: List[A], protected val out: L
     *  @return the first element of the queue.
     */
   def dequeue: (A, Queue[A]) = out match {
-    case Nil if !in.isEmpty => val rev = in.reverse ; (rev.head, new Queue(Nil, rev.tail))
+    case Nil if !in.isEmpty => in match {
+      case x :: Nil => (x, new Queue(Nil, Nil))
```

Comment:
I'm not sure how much this line adds. The cost of reversing a `List` with a single element is pretty low, and ideally doesn't happen very often (whereas this check has to be done for all reversals). Without benchmarks showing it's an improvement for near-empty `Queue`s and not a significant cost for larger ones, I'm not convinced this check improves things.

Additionally, in order to fulfill the optimisation requested by the linked ticket, there should be a check for `out` having a single element, in which case `in` is eagerly reversed (slightly before it's needed, but presumably it's going to be removed eventually anyway - it's a queue after all).

Reply:
(@gi114 is new to the repo, so might need pointers on running benchmarks if we decide that's needed here)

Reply:
Thanks for your comments, I shall try to adjust the PR. And yes, this is my first ticket so let me know if I there is anything relevant I should do. Also, do you usually run tests suites or do you test your changes separately on a ./gitignore file?